AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -168,9 +198,54 @@ - (void)didLoad
         dispatch_async(dispatch_get_main_queue(), ^{
           _placeholderImageNode.frame = self.bounds;
           [self insertSubnode:_placeholderImageNode atIndex:0];
+          // [self setNeedsLayout];
         });
-      }];
-    });
+      }
+    }];
+  });
+}
+
+- (void)interfaceStateDidChange:(ASInterfaceState)newState fromState:(ASInterfaceState)oldState
+{
+  if (!(newState & ASInterfaceStateVisible)) {
+    if (oldState & ASInterfaceStateVisible) {
+      if (_shouldBePlaying) {
+        [self pause];
+        _shouldBePlaying = YES;
+      }
+      [(UIActivityIndicatorView *)_spinner.view stopAnimating];
+      [_spinner removeFromSupernode];
+    }
+  } else {
+    if (_shouldBePlaying) {
+      [self play];
+    }
+  }
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context
+{
+  if ([[change objectForKey:@"new"] integerValue] == AVPlayerItemStatusReadyToPlay) {
+    if ([self.subnodes containsObject:_spinner]) {
+      [_spinner removeFromSupernode];
+      _spinner = nil;
+    }
+    
+    // If we don't yet have a placeholder image update it now that we should have data available for it
+    if (!_placeholderImageNode) {
+      if (_currentPlayerItem &&
+          _currentPlayerItem.tracks.count > 0 &&
+          _currentPlayerItem.tracks[0].assetTrack &&
+          _currentPlayerItem.tracks[0].assetTrack.asset) {
+        _asset = _currentPlayerItem.tracks[0].assetTrack.asset;
+        [self setPlaceholderImagefromAsset:_asset];
+        [self setNeedsLayout];
+      }
+    }
+  }
+  
+  if ([[change objectForKey:@"new"] integerValue] == AVPlayerItemStatusFailed) {
```

Comment:
I'm not really sure what the @"new" key is for, but can we cache this in a local variable at the top rather than accessing the dictionary and unpacking the integer twice?